build/xunit.runsettings

```diff
@@ -1,20 +1,44 @@
+
+      The default value is 1, meaning that only one testhost runs at the same time. A special value 0 allows as many testhosts as you have logical processors (for example, 6, for a computer with 6 physical cores without multi-threading, or 12, for a computer with six physical cores with multi-threading).
+
+      The number of distinct DLLs in the run determines the actual number of testhosts started.
+    -->
+    <MaxCpuCount>4</MaxCpuCount>
+
+    <!-- Specify a Boolean value, which defines the exit code when no tests are discovered. If the value is true and no tests are discovered, a nonzero exit code is returned. Otherwise, zero is returned. -->
+    <TreatNoTestsAsError>true</TreatNoTestsAsError>
+
+    <EnvironmentVariables>
+      <!-- Changes the default timeout in VSTest for shutdown to 1000ms instead of 100ms which can cause intermittent failures in CI. -->
+      <VSTEST_TESTHOST_SHUTDOWN_TIMEOUT>1000</VSTEST_TESTHOST_SHUTDOWN_TIMEOUT>
+    </EnvironmentVariables>
+  </RunConfiguration>
+
+  <!-- https://xunit.net/docs/runsettings -->
+  <xUnit>
+    <!-- Set this value to true to include diagnostic information during test discovery and execution. Each runner has a unique method of presenting diagnostic messages. -->
+    <DiagnosticMessages>true</DiagnosticMessages>
+
+    <!-- Set this value to enable long-running (hung) test detection. When the runner is idle waiting for tests to finished, it will report that fact once the timeout has passed. Use a value of 0 to disable the feature, or a positive integer value to enable the feature (time in seconds). -->
+    <LongRunningTestSeconds>120</LongRunningTestSeconds>
+
+    <!-- Set this to override the maximum number of threads to be used when parallelizing tests within this assembly. Use a value of 0 or "default" to indicate that you would like the default behavior; use a value of -1 or "unlimited" to indicate that you do not wish to limit the number of threads used for parallelization. -->
+    <MaxParallelThreads>4</MaxParallelThreads>
+
+    <!-- Set this to true if this assembly is willing to participate in parallelization with other assemblies. Test runners can use this information to automatically enable parallelization across assemblies if all the assemblies agree to it. -->
+    <ParallelizeAssembly>false</ParallelizeAssembly>
+
+    <!-- Set this to true if the assembly is willing to run tests inside this assembly in parallel against each other. Tests in the same test collection will be run sequentially against each other, but tests in different test collections will be run in parallel against each other. Set this to false to disable all parallelization within this test assembly. -->
+    <ParallelizeTestCollections>false</ParallelizeTestCollections>
```

Comment:
I compared the CI run for this PR to a different CI run, because I would have expected that setting ParallelizeTestCollections and ParallelizeAssembly both to false would make the tests run very slowly. But somehow this PR's CI run was margionally faster then the dev branch run.

It's totally not intuitive to me why, but I didn't see this mentioned in the PR description or anywhere else, so I wanted to point out that apparently the CI is not going to take a long longer with this PR merged.

Reply:
I struggle to understand it as well. Its still running tests assemblies in parallel with each other, but I guess not running tests themselves in parallel?  A machine with only 2 cores would only be able to run two tests at the same time anyway so I think the adjustments don't affect the maximum parallelization we could already achieve on CI agents.  It really just lowers the amount of parallelization that's queued up waiting for CPU.